pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -980,3 +952,93 @@ func TestCrossValidateNetworkAndGnp(t *testing.T) {
 	}
 
 }
+
+func TestHandleGKENetworkParamSetDelete_NetworkInUse(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	ctx, stop := context.WithCancel(context.Background())
+	defer stop()
+
+	testVals := setupGKENetworkParamSetController(ctx)
+	testVals.runGKENetworkParamSetController(ctx)
+
+	networkName := "test-network"
+	gkeNetworkParamSetName := "test-paramset"
+
+	network := &networkv1.Network{
+		ObjectMeta: v1.ObjectMeta{
+			Name: networkName,
+		},
+	}
+
+	paramSet := &networkv1.GKENetworkParamSet{
+		ObjectMeta: v1.ObjectMeta{
+			Name:       gkeNetworkParamSetName,
+			Finalizers: []string{GNPFinalizer},
+		},
+		Status: networkv1.GKENetworkParamSetStatus{
+			NetworkName: networkName,
+		},
+	}
+
+	_, err := testVals.networkClient.NetworkingV1().Networks().Create(ctx, network, v1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Network: %v", err)
+	}
+
+	_, err = testVals.networkClient.NetworkingV1().GKENetworkParamSets().Create(ctx, paramSet, v1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create GKENetworkParamSet: %v", err)
+	}
+
+	// Test case: Network is in use
+	network.SetAnnotations(map[string]string{
+		networkv1.NetworkInUseAnnotationKey: networkv1.NetworkInUseAnnotationValTrue,
+	})
+	_, err = testVals.networkClient.NetworkingV1().Networks().Update(ctx, network, v1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to update Network status: %v", err)
+	}
+
+	paramSet, err = testVals.networkClient.NetworkingV1().GKENetworkParamSets().Get(ctx, gkeNetworkParamSetName, v1.GetOptions{})
+	if err != nil {
+		t.Error(err)
+	}
+	now := v1.Now()
+	paramSet.SetDeletionTimestamp(&now)
+	_, err = testVals.networkClient.NetworkingV1().GKENetworkParamSets().Update(ctx, paramSet, v1.UpdateOptions{})
+	if err != nil {
+		t.Error(err)
+	}
+
+	g.Consistently(func() (bool, error) {
+		return testVals.doesGNPFinalizerExist(ctx, gkeNetworkParamSetName)
+	}).Should(gomega.BeTrue(), "finalizer should not be removed from GKENetworkParamSet")
+
+	// Test case: Network is not in use
+	network.SetAnnotations(map[string]string{})
+	_, err = testVals.networkClient.NetworkingV1().Networks().Update(ctx, network, v1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to update Network status: %v", err)
+	}
+
+	g.Eventually(func() (bool, error) {
+		return testVals.doesGNPFinalizerExist(ctx, gkeNetworkParamSetName)
+	}).Should(gomega.BeFalse(), "finalizer should be removed from GKENetworkParamSet")
+}
+
+func (testVals *testGKENetworkParamSetController) doesGNPFinalizerExist(ctx context.Context, gkeNetworkParamSetName string) (bool, error) {
+	paramSet, err := testVals.networkClient.NetworkingV1().GKENetworkParamSets().Get(ctx, gkeNetworkParamSetName, v1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	finalizerExists := false
```

Comment:
you don't need the bool, just return true, nil if is found and `false, nil` if you go out of the loop

Reply:
done